nomenclature/cli.py

```diff
@@ -93,3 +96,31 @@ def cli_valid_project(
     """
     assert_valid_yaml(path)
     assert_valid_structure(path, definitions, mappings, required_data, dimensions)
+
+
+@cli.command("run-region-processing")
+@click.argument("input_data_file", type=click.Path(exists=True, path_type=Path))
+@click.option(
+    "-d",
+    "--definitions",
+    type=click.Path(exists=True, path_type=Path),
+    default="definitions",
+)
+@click.option(
+    "-m", "--mappings", type=click.Path(exists=True, path_type=Path), default="mappings"
+)
+@click.option("--export-processing-result", is_flag=True, default=True)
+@click.option("--export-differences", is_flag=True, default=False)
+def run_region_processing(
+    input_data_file: Path,
+    definitions: Path,
+    mappings: Path,
+    export_processing_result: bool,
+    export_differences: bool,
+):
+
+    result = RegionProcessor.from_directory(
```

Comment:
I suggest to add a validation against the variable code-list just to be safe.

Reply:
You mean a validation of the input data right? The `RegionProcessor` already gets validated.

Reply:
No, I mean validating the variables in the IamDataFrame (or maybe validate as part of RegionProcessor.apply`?). We don't want to perform region-processing without having explicit rules for aggregation (where having a code without region-aggregation-attributes is explicitly using a summation).

Reply:
Ok sounds good, I'll add that

Reply:
Switched to using `core.process` so that variable validation is included.